### PR TITLE
Improve "cartservice" demo ability

### PR DESCRIPTION
**Pull request by pjanotti**
Add some env var to control behavior of the cartservice that can be used to demo features of Splunk Observability.

Please, don't "fix" this code 😬 

Some links and images of the CPU and blocking optimizations (allocation needs to wait the actual feature):

- These are stacks from excessive CPU usage, look for Bogo.Sort, cartservice, Redis for stacks directly from the server https://app.signalfx.com/#/profiling?endTime=Now&environments=tmp-test-dotnet-demo&onlyWithTraces=1&service=cartservice&startTime=-1d
- Disabling the excessive CPU issue above we get stacks from excessive blocking (notice the change in environment): https://app.signalfx.com/#/profiling?endTime=Now&environments=tmp-test-dotnet-demo-00&onlyWithTraces=1&service=cartservice&startTime=-1d

These two above are all with 10 second interval collection and very low load so I assume that when we actually deploy it we are going to burn some CPU and have many valid stacks for spans. 

CPU usage after excessive CPU usage is disabled: 
image

Number of threads after contention issue is disabled:
image

### CODE REVIEW

File: src/cartservice/cartstore/Bogo.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+  public static class Bogo
+  {
+    public static void Sort(List<char> list)
+    {
+      Shuffle(list);
+      while (!IsSorted(list))
+      {
+        Shuffle(list);
+      }
```
Comment (pellared) [inline src/cartservice/cartstore/Bogo.cs:14]:
```suggestion
      do 
      {
        Shuffle(list);
      } while (!IsSorted(list));
```

**Review by pellared — APPROVED** [review]
@aurbiztondo-splunk I think it may be good if you take a look 😉

File: src/cartservice/README.md [hunk 2]
```diff
@@ -1,34 +1,53 @@
...
+- `EXTERNAL_DB_NAME`: string, name of external database
+
+### Optimizations
+
+These environment variables are used to trigger behaviors on the application that affect its perfomance characteristics.
+Each optimization affects lantency, CPU, and memory in different ways.
+Keep that in mind when interpreting their effect.
+All these environment variables are boolean.
+
+- `FIX_EXCESSIVE_ALLOCATION`: bool, controls if the excessive allocation per requests should be fixed or not
```
Comment (pellared) [inline src/cartservice/README.md:25]:
```suggestion
- `FIX_EXCESSIVE_ALLOCATION`: bool, controls if the excessive allocation per request should be happening or not
```

File: src/cartservice/README.md [hunk 3]
```diff
@@ -1,34 +1,53 @@
...
+- `FIX_SLOW_LEAK`: bool, controls if a slow leak is going to be happening or not
```
Comment (pellared) [inline src/cartservice/README.md:26]:
```suggestion
- `FIX_SLOW_LEAK`: bool, controls if a slowly growing memory leak is going to be happening or not
```

File: src/cartservice/README.md [hunk 4]
```diff
@@ -1,34 +1,53 @@
...
+- `OPTIMIZE_CPU`: bool, controls if CPU is going to be used in a bad input validation code
```
Comment (pellared) [inline src/cartservice/README.md:27]:
```suggestion
- `OPTIMIZE_CPU`: bool, controls if  the CPU is going to be efficiently during input validation
```

Comment (pjanotti) [reply src/cartservice/README.md:27]:
```suggestion
- `OPTIMIZE_CPU`: bool, controls if  the CPU is going to be efficiently used during input validation

```

**Review by dmitryax — APPROVED** [review]

**Pull request merged**